Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+                              constraints: constraints),
+            
+            ListUpdateRequest(threatType: .unwantedSoftware,
+                              platformType: .any,
+                              threatEntryType: .url,
+                              state: database.getState(.unwantedSoftware),
+                              constraints: constraints)
+        ]
+        
+        do {
+            let body = FetchRequest(client: clientInfo, listUpdateRequests: lists)
+            let request = try encode(.post, endpoint: .fetch, body: body)
+            executeRequest(request, type: FetchResponse.self) { [weak self] response, error in
+                guard let self = self else { return }
+                
+                if let error = error {
+                    return completion(error)
+                }
+                
+                if let response = response {
+                    var didError = false
+                    self.database.update(response, completion: {
+                        if let error = $0 {
+                            log.error("Safe-Browsing: Error Updating Database: \(error)")
+                            didError = true
+                        }
+                    })
+                    
+                    return completion(didError ? SafeBrowsingError("Safe-Browsing: Error Updating Database") : nil)
+                }
+                
+                completion(nil)
+            }
+        } catch {
+            completion(error)
+        }
+    }
+    
+    private func encode<T>(_ method: RequestType, endpoint: Endpoint, body: T) throws -> URLRequest where T: Encodable {
+        
+        let urlPath = "\(baseURL)\(endpoint.rawValue)?key=\(SafeBrowsingClient.apiKey)"
+        
+        guard let url = URL(string: urlPath) else {
+            throw SafeBrowsingError("Invalid Request")
+        }
+        
+        var request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData)
+        request.httpMethod = method.rawValue
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.setValue("application/json", forHTTPHeaderField: "Accept")
+        request.httpBody = try JSONEncoder().encode(body)
+        
+        return request
+    }
+    
+    @discardableResult
+    private func executeRequest<T>(_ request: URLRequest, type: T.Type, completion: @escaping (T?, Error?) -> Void) -> URLSessionDataTask where T: Decodable {
+        let task = session.dataTask(with: request) { data, response, error in
+            if let error = error {
+                return completion(nil, error)
+            }
+            
+            guard let data = data else {
+                return completion(nil, SafeBrowsingError("Invalid Server Response: No Data"))
+            }
+            
+            if let response = response as? HTTPURLResponse {
+                if response.statusCode < 200 || response.statusCode > 299 {
```

Comment:
why not just check for 200/301 here?

Reply:
The [spec says](https://developers.google.com/safe-browsing/v4/request-frequency#back-off-mode) that we need to return an error for any non-`200` status codes so this can simply be `if response.statusCode != 200`.

Reply:
Fixed.